Mathlib/Logic/Function/DependsOn.lean

```diff
@@ -0,0 +1,64 @@
+/-
+Copyright (c) 2024 Etienne Marion. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Etienne Marion
+-/
+import Mathlib.Data.Set.Function
+
+/-!
+# Functions depending only on some variables
+
+When dealing with a function `f : Π i, α i` depending on many variables, some operations
+may get rid of the dependency on some variables (see `Function.updateFinset` or
+`MeasureTheory.lmarginal` for example). However considering this new function
+as having a different domain with fewer points is not comfortable in lean, as it requires the use
+of subtypes and can lead to tedious writing.
```

Comment:
I agree this is not comfortable, but sometimes it is necessary: can we have the lemma that `DependsOn` is equivalent to the existence of this function?